src/lib/components/Icon/Icon.svelte

```diff
@@ -1,31 +1,27 @@
+<svelte:options immutable />
+
 <script lang="ts">
   import classNames from "$lib/util/classNames";
-  import icons from "./icons";
-  type IconName = Extract<keyof typeof icons, string>;
 
-  export let name: IconName;
   // Size is taken from the USWDS implemention of icons in their icon component package (see
   // https://designsystem.digital.gov/components/icon/#using-the-icon-component-2 for more details).
   // Reason for undefined `size` value to start: "By default, icons will scale with font size.
   // If you want to specify an icon size, use one of the component’s size variants."
   export let size: 3 | 4 | 5 | 6 | 7 | 8 | 9 | undefined = undefined;
-  export let className = "";
 
-  // As a default value, make the alt just a more titleized version of the file name,
-  // e.g., "arrow_upward" becomes "Arrow Upward"
-  export let alt = name
-    .split("_")
-    .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
-    .join(" ");
```

Comment:
Is there another way we can provide a default here? Could we do something similar to what we had here by extracting the icon name out of the `src` string?

Reply:
I don't think providing a default `alt` is a good idea to begin with. The names of the icons vary a lot, so making sure that all of the autogenerated labels are meaningful is already going to be difficult, and the choice of alt text for an icon is very situational. For instance, if there is no accompanying text, the icon should have alt text describing what the icon does, not what it visually looks like. Another example would be something like the search icon which appears right next to a "Search" label and so should have `alt=""`.

Reply:
Right, good point. For some reason I thought we had added the default `alt` due to guidance from the USWDS but that doesn't seem to be the case. You're right that we don't always want an `alt` attribute on these images. Disregard then!